src/main/scala/com/vngrs/etl/Combine.scala

```diff
@@ -0,0 +1,49 @@
+package com.vngrs.etl
+
+import org.apache.spark.rdd.RDD
+
+import scala.reflect.ClassTag
+
+/**
+  * [[com.vngrs.etl.Combine]] is an interface for combining two [[com.vngrs.etl.Pipeline]]s into one.
+  *
+  * @tparam A Type of Input [[com.vngrs.etl.Pipeline]]
+  * @tparam B Type of Output [[com.vngrs.etl.Pipeline]]
+  */
+trait Combine[A, B] {
```

Comment:
Okay, let's stick with the plan and improve design later. But for now, at least refactor trait to an abstract class so it's intention will be clearer.

```scala
abstract class Combine[A, B, C](pipeline1: Pipeline[A]) {
  def apply(pipeline2: Pipeline[B]): Pipeline[C]
}
```